### PR TITLE
Feature/reconsent feedback screen

### PR DESCRIPTION
**Pull request by aikewoody**
# Description

https://www.notion.so/joinzoe/017c3de591604a7b89d0b69e3c37b279?v=0133357df3ed42e1bd42c1e0d7afcf45&p=0f357b7ae1a14746996b4cff08ddb4e3

https://www.figma.com/file/7TEcX7f6rw3Z0WYR4TpX3R/COVID-Symptom-Study-(App)---MASTER?node-id=11581%3A153634

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

Simulator Screen Shot - iPhone 11 - 2021-07-05 at 16 21 06

Simulator Screen Shot - iPhone 11 - 2021-07-05 at 16 21 15

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if needed)

## Checklist (for reviewers)

- [x] Checked against Figma designs (if relevant)
- [x] Checked that data has been successfully saved to the backend (if relevant)

## Out of scope and potential follow-up

_Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?_

### CODE REVIEW

**Review by juliatan — APPROVED** [review]

File: assets/icons/Check.tsx [hunk 1]
```diff
@@ -2,23 +2,22 @@ import { colors } from '@theme/colors';
 import * as React from 'react';
 import Svg, { Path } from 'react-native-svg';
 
-type CheckProps = {
-  width?: number | string;
+type IProps = {
```
[the remark this answers is not in the document]
Comment (aikewoody) [reply assets/icons/Check.tsx:5]: Sharp eye!

File: src/features/reconsent/screens/ReconsentFeedbackScreen.tsx [hunk 2]
```diff
@@ -1,34 +1,97 @@
...
+
+  async function onPress() {
+    setLoading(true);
+    try {
+      const context = options.reduce((previous, current) => {
+        previous[current.id] = current.value;
+        return previous;
+      }, {} as IContext);
+      await generalApiClient.postUserEvent('feedback_reconsent', context);
+    } catch (_) {}
```
[the remark this answers is not in the document]
Comment (aikewoody) [reply src/features/reconsent/screens/ReconsentFeedbackScreen.tsx:70]: Let me ask product/design about this

File: src/components/inputs/CheckboxTextInputList.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+          >
+            <View style={props.dark ? styles.checkBoxDark : styles.checkBox}>
+              {option.value !== undefined ? <Check color={colors.purple} /> : null}
+            </View>
+            <Text rhythm={16} style={styles.text} textClass="pLight">
+              {option.label}
+            </Text>
+          </TouchableOpacity>
+          {option.value !== undefined ? (
+            <Textarea
```
[the remark this answers is not in the document]
Comment (aikewoody) [reply src/components/inputs/CheckboxTextInputList.tsx:50]: The design doesn't show this, can ask Solveig about it

File: src/features/reconsent/screens/ReconsentFeedbackScreen.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (aikewoody) [reply src/features/reconsent/screens/ReconsentFeedbackScreen.tsx:70]: Actually this logic will be moved to the thank you screen

**Pull request merged**